### DISCUSSION TITLE
Set simulated load junctions / batteries

### DISCUSSION THREAD

**Discussion by Ek-Videogames** [FakeFishGames/Barotrauma#10306] [Ideas and suggestions]
Junction boxes can output their power & load values, but there's still not much that can be done with that information to react & fine-tune electrical systems. What I'm suggesting is a few possible ideas for inputs, which can alter the 'perceived' load on the grid - thereby causing the reactor to make more/less power, or batteries/supercapacitors to output more/less power than the 'real' load requires.

- "load_add" offset the junction's natural load calculation by this amount.
_(Fine tuning certain systems to purposely run high if expecting a large load jump, 
or purposely undervolting the grid to save power when idle.)_

- possibly "load_scale" which is similar to add but multiplicative, IE `1000 * 0.5 = 500` `1000 * 2.0 = 2000`
although this could also be done by taking the vanilla load_out pin, doing some math, then using "load_add"

- "load_set" overrides the junction's natural load calculation with a value.
_(I don't immediately know how this would be useful but somebody probably has an idea. 
Maybe on subs where you expect a constant power usage, or for implementing a 'silent running' mode)_

**Comment by MapleWheels** [comment]
This can already be achieved via circuitry in the game. Literally just an ADD/SUB component. Anything involving Reactor Control in this fashion is not only already achievable, but staple to many "boost" and "eco" functions in submarines.

**Reply by Ek-Videogames** [reply]
> This can already be achieved via circuitry in the game. Literally just an ADD/SUB component. Anything involving Reactor Control in this fashion is not only already achievable, but staple to many "boost" and "eco" functions in submarines.

I'm not sure I follow - which junction pin is this?

**Reply by MapleWheels** [reply]
> > This can already be achieved via circuitry in the game. Literally just an ADD/SUB component. Anything involving Reactor Control in this fashion is not only already achievable, but staple to many "boost" and "eco" functions in submarines.
> 
> I'm not sure I follow - which junction pin is this?

What he is saying is that he wants to be able to make the Reactor "think" that there is more load than there actually is. My response was that his end goal was already doable with reactor control circuits, except it's just over setting power vs load. Batteries could benefit from this I guess.

**Reply by meelix** [reply]
That's not what he meant. The solution you are recommending is just basic over/under-volting of the entire sub.
Being able to adjust the load a specific junction box provides would allow to over-volt specific parts of the ship and/or under-volt at the same time

**Comment by meelix** [comment]
I'd love this, but it would require a rework of the junction boxes, since right now a junction box just distributes the same power across all the connected wires it would increase the load for the entire sub rather then just the parts you want.

An explicit in and out pin would solve this but it would mess up all the wiring prior to this change (and allow exploiting it)

I would really welcome this addition if implemented properly. Although just the set_load pin so you'd still have to do the logic and math yourself

**Reply by meelix** [reply]
Just noticed it probably wouldn't work anyways. Looking at a sub like the orca 2 where all the junction boxes are connected in series. Say you set junction box nr. 3 to over-volt by 25% and junction box nr. 4 to under-volt by 50%... where would the energy go? Same vice versa, you cant just get more power without the previous junction box providing more  then that or at least the same.

It would likely also require a lot of other reworks to other properly working systems
